templates/approval/deployment.yaml

```diff
@@ -0,0 +1,78 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "incore.fullname" . }}-approval
+  labels:
+    {{- include "incore.labels" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.approval.replicaCount }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "incore.name" . }}-approval
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "incore.name" . }}-approval
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      {{- with .Values.approval.image.pullSecrets | default .Values.image.pullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: approval
+          image: "{{ .Values.image.project }}/{{ .Values.approval.image.repository }}:{{ .Values.approval.image.tag }}"
+          imagePullPolicy: {{ .Values.approval.image.pullPolicy | default .Values.image.pullPolicy }}
+          env:
+            - name: ADMIN_LIST
+              value: {{ .Values.approval.ADMIN_LIST | quote }}
+            - name: KEYCLOAK_USERNAME
+              value: {{ .Values.approval.KEYCLOAK_USERNAME | quote }}
+            - name: KEYCLOAK_PASSWORD
+              {{- if .Values.approval.KEYCLOAK_PASSWORD }}
```

Comment:
let not put the direct insert of KEYCLOAK password in here, instead just make it only to use the secret like you put below

Reply:
but I would need to overwrite it with production keycloak password on dev, hence I have this additional logic.

Reply:
dev and prod uses different secret so you can set it in there, instead of directly put